src/classes/INT_InteractionProcessor.cls

```diff
@@ -307,57 +302,52 @@ public class INT_InteractionProcessor {
      * @param leads, a List of Lead objects to insert.
      * @return leadsToConvert, a List of Database.LeadConvert records to convert.
      */
-    private List<Database.LeadConvert> insertLeads(List<Lead> leadsToInsert) {
+    private List<Database.LeadConvert> insertLeads(List<Lead> leadsToInsert, List<Interaction__c> interactions) {
```

Comment:
@karamcintyre the main change Jim did is to this whole method here. You can see on L315 is where he changed how the `SaveResult` records are inserted and evaluated. Instead of inserting them and assuming the LeadId was present (how it was) we not loop through them and check to ensure duplicates are found and handled properly.